source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -0,0 +1,325 @@
+
+    for (const string& target : targets)
+    {
+        if (_ocioInfo->languageMap.count(target))
+        {
+            _ocioInfo->language = static_cast<int>(OCIO::GPU_LANGUAGE_GLSL_4_0);
+            _ocioInfo->target = target;
+            break;                
+        }
+    }
+}
+
+bool OCIOColorManagementSystem::setConfigFile(const FilePath& configFile)
+{
+    if (_configFile != configFile && configFile.size())
+    {
+        _configFile = configFile;
+        _colorSpaceNames.clear();
+
+        _ocioInfo->config = OCIO::Config::CreateFromFile(_configFile.asString().c_str());
+        if (!_ocioInfo->config)
+        {
+            OCIO::GetCurrentConfig();
+        }
+        if (_ocioInfo->config)
+        {
+            OCIO::ColorSpaceSetRcPtr colorSpaces = _ocioInfo->config->getColorSpaces(nullptr);
+            int colorSpaceCount = colorSpaces->getNumColorSpaces();
+            for (int i = 0; i < colorSpaceCount; i++)
+            {
+                _colorSpaceNames.insert(colorSpaces->getColorSpaceNameByIndex(i));
+            }
+            return true;
+        }
+    }
+    return false;
+}
+
+/// Returns whether this color management system supports a provided transform
+bool OCIOColorManagementSystem::supportsTransform(const ColorSpaceTransform& transform) const
+{
+    if (_ocioInfo->language == OCIO_UNSUPPORT_LANGUAGE)
+    {
+        return false;
+    }
+
+    OCIO::ConstProcessorRcPtr processor = _ocioInfo->config->getProcessor(transform.sourceSpace.c_str(), transform.targetSpace.c_str());
+    return (nullptr != processor);
+}
+
+/// Create a node to use to perform the given color space transformation.
+ShaderNodePtr OCIOColorManagementSystem::createNode(const ShaderGraph* parent, const ColorSpaceTransform& transform, const string& name, GenContext& context) const
+{
+    if (transform.type != Type::COLOR3 && transform.type != Type::COLOR4)
+    {
+        throw ExceptionShaderGenError("Invalid type specified to color transform: '" + transform.type->getName() + "'");
+    }
+
+    ImplementationPtr impl = getImplementation(transform);
+
+    if (!impl)
+    {
+        throw ExceptionShaderGenError("No implementation found for transform: ('" + transform.sourceSpace + "', '" + transform.targetSpace + "').");
+    }
+
+    // Check if it's created and cached already, otherwise create and cache it.
+    ShaderNodeImplPtr nodeImpl = context.findNodeImplementation(impl->getName());
+    if (!nodeImpl)
+    {
+        // Note: Use custom OCIO "source code node".
+        nodeImpl = OCIOSourceCodeNode::create();
+        nodeImpl->initialize(*impl, context);
+        context.addNodeImplementation(impl->getName(), nodeImpl);
+    }
+
+    // Create the node.
+    ShaderNodePtr shaderNode = ShaderNode::create(parent, name, nodeImpl, ShaderNode::Classification::TEXTURE);
+
+    // Create ports on the node.
+    ShaderInput* input = shaderNode->addInput("in", transform.type);
```

Comment:
todo: check if input and output shader port names need to match the source code generation.